cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AdapterHttpHandler.java

```diff
@@ -276,7 +276,7 @@ public void deleteAdapter(HttpRequest request, HttpResponder responder,
       adapterService.removeAdapter(Id.Namespace.from(namespaceId), adapterName);
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (CannotBeDeletedException e) {
-      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
+      responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
```

Comment:
This is what I see so far:

```
$ gj "catch.*\\(CannotBeDeletedException.*\)" -A2 *
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AdapterHttpHandler.java:    } catch (CannotBeDeletedException e) {
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AdapterHttpHandler.java-      responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AdapterHttpHandler.java-    } catch (NotFoundException e) {
--
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java:      } catch (CannotBeDeletedException cbde) {
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java-        appStatus = AppFabricServiceStatus.PROGRAM_STILL_RUNNING;
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java-      } catch (Exception e) {
--
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java:    } catch (CannotBeDeletedException cbde) {
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java-      status = AppFabricServiceStatus.PROGRAM_STILL_RUNNING;
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java-    } catch (Exception e) {
--
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java:    } catch (NamespaceCannotBeDeletedException e) {
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java-      responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java-    } catch (Exception e) {
--
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java:    } catch (NamespaceCannotBeDeletedException e) {
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java-      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java-    } catch (Exception e) {
```

So its a mix of ``FORBIDDEN`` (delete app, delete namespaces - which has one ``FORBIDDEN`` and another ``CONFLICT``) and ``CONFLICT`` currently. If we were to change everything to ``CONFLICT``, we'd have two API changes. I am fine with standardizing all of these in this PR.

Thoughts?